File: SampleGCDTable/SampleGCDTable/ViewController.m
//
//  ViewController.m
//  SampleGCDTable
//
//  Created by Chinna Venkata Raju A on 9/26/14.
//  Copyright (c) 2014 Paradigm Creatives. All rights reserved.
//

#import "ViewController.h"

@interface ViewController (){
    NSMutableData *imagesData;
    NSMutableArray *imagesArray;
}
@property (weak, nonatomic) IBOutlet UITableView *imagesTableView;

@end

@implementation ViewController


-(void)parseJsonObject:(NSDictionary*)responseDict{
    NSLog(@"response:%@",responseDict);
    NSArray *pages = [responseDict objectForKey:@"pages"];
    for (NSDictionary *imageBox in pages) {
        NSArray *imageboxes = [imageBox objectForKey:@"ImageBoxArray"];
        for (NSDictionary *imageBox in imageboxes) {
            NSString *imageStr = [imageBox objectForKey:@"Image"];
            NSLog(@"imageStr:%@",imageStr);
            [imagesArray addObject:imageStr];
        }
    }
    [self.imagesTableView reloadData];
}

#pragma mark NSURLConnection Delegate
-(void)connection:(NSURLConnection *)connection didFailWithError:(NSError *)error{
    NSLog(@"error:%@",[error localizedDescription]);
}

-(void)connection:(NSURLConnection*)connection didReceiveResponse:(NSURLResponse *)response{
    NSLog(@"got response");
}

-(void)connection:(NSURLConnection*)connection didReceiveData:(NSData *)data{
    [imagesData appendData:data];
}

- (void)connectionDidFinishLoading:(NSURLConnection *)connection{
    NSError *error = nil;
    id jsonObj = [NSJSONSerialization JSONObjectWithData:imagesData options:0 error:&error];
    NSDictionary *responseDict = (NSDictionary*)jsonObj;
    [self parseJsonObject:responseDict];
}


- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    imagesData = [[NSMutableData alloc] init];
    imagesArray = [[NSMutableArray alloc]init];
    
    NSURL *imagesUrl = [NSURL URLWithString:@""];
    NSURLRequest *urlReq = [NSURLRequest requestWithURL:imagesUrl];
    NSURLConnection *urlConnection = [[NSURLConnection alloc] initWithRequest:urlReq delegate:self];
    [urlConnection start];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark TableViewDataSource
-(NSInteger)numberOfSectionsInTableView:(UITableView *)tableView{
    return 1;
}

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return imagesArray.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"myImage"];
    NSString *imageString = [imagesArray objectAtIndex:indexPath.row];
    
    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), ^{
        NSURL *imageUrl = [NSURL URLWithString:imageString];
        NSData *imagedata=[NSData dataWithContentsOfURL:imageUrl];
        dispatch_async(dispatch_get_main_queue(), ^{
            [cell.imageView setImage:[UIImage imageWithData:imagedata]];
        });
    });

    
//    NSData *imageData = [NSData dataWithContentsOfURL:imageUrl];
//    NSLog(@"imageData:%@",imageData);
//    UIImage *myImage = [UIImage imageWithData:imageData];
//    cell.imageView.image = myImage;
    return cell;
}
@end

File: JsonLink/JsonLink/ViewController.swift
//
//  ViewController.swift
//  JsonLink
//
//  Created by mac mini on 11/4/14.
//  Copyright (c) 2014 ParadigmCreatives. All rights reserved.
//

import UIKit

class ViewController: UIViewController,NSURLConnectionDelegate{
    
    var data = NSMutableData()
    
    override func viewDidLoad() {
        super.viewDidLoad()
       self.startConnection()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
 func startConnection()
 {
    let filePath:NSString = "http://api.dan.co.jp/jsonenv"
    var url:NSURL = NSURL(string: filePath)
    var request:NSURLRequest = NSURLRequest(URL: url)
    var connection:NSURLConnection = NSURLConnection(request: request, delegate: self)
    connection.start()

 }
    func connection(connection: NSURLConnection, didReceiveData data: NSData)
    {
        self.data.appendData(data)
    }

    func connectionDidFinishLoading(connection: NSURLConnection)
    {
        var error:NSError?
        var jsonObject:NSDictionary = NSJSONSerialization.JSONObjectWithData(data, options: NSJSONReadingOptions.MutableContainers, error: nil) as NSDictionary
        
                //println(jsonObject)
        
        let value = jsonObject["DOCUMENT_ROOT"] as NSString
        println(value)
        
        
        var data1:NSData = NSJSONSerialization.dataWithJSONObject(jsonObject, options: NSJSONWritingOptions(), error: nil)!
        
        var filepath:NSString = "/Users/mac mini/Desktop/Jsongenerated.json"
        var outputStream:NSOutputStream = NSOutputStream.outputStreamToFileAtPath(filepath, append: false)
        outputStream.open()

        NSJSONSerialization.writeJSONObject(jsonObject, toStream: outputStream, options:NSJSONWritingOptions(), error: nil)
    }

}

